File: from-3401-to-3500/3442-maximum-difference-between-even-and-odd-frequency-i/solution.go
func maxDifference(s string) int {
    freq := make(map[rune]int)
    for _, c := range s {
        freq[c] += 1
    }

    maxOdd := 0
    minEven := len(s) + 1
    for _, f := range freq {
        if f % 2 == 1 && f > maxOdd {
            maxOdd = f
        } else if f % 2 == 0 && f < minEven {
            minEven = f
        }
    }
    return maxOdd - minEven
}

File: from-3401-to-3500/3442-maximum-difference-between-even-and-odd-frequency-i/solution.py
class Solution:
    def maxDifference(self, s: str) -> int:
        freq = [0] * 26
        for c in s:
            freq[ord(c) - ord('a')] += 1

        max_odd = 0
        min_even = len(s) + 1
        for f in freq:
            if f > 0:
                if f % 2 and f > max_odd:
                    max_odd = f
                elif f % 2 == 0 and f < min_even:
                    min_even = f

        print(max_odd, min_even)
        return max_odd - min_even
